### ISSUE TITLE
training custom dataset problem

### ISSUE THREAD

**Issue by betterhalfwzm** [thuyngch/ATSS-EfficientDet-PyTorch#5]
I trained my dataset on eficientnet B0. Now I have trained 77 epochs. I test the latest model with a threshold of 0.1. For example, the following pictures will be missed. Is that because the number of iterations is not enough? Can you give me some suggestions? Thank you very much！！！

![image](https://user-images.githubusercontent.com/20449990/79712085-6967c800-82fc-11ea-9522-9b233df591c3.png)

config：
# Training
total_epochs = 300
imgs_per_gpu = 16

#lr_start = 8e-2
lr_start = 4e-2
lr_end = 1e-4
warmup_iters = 917
warmup_ratio = 1e-5
weight_decay = 4e-5

log_interval = 50
ckpt_interval = 1
evaluation = dict(interval=2, metric='bbox')

# Dataset
img_scale = (512, 512)
keep_ratio = False
size_divisor = 32
repeat_times = 1
workers_per_gpu = 4
data_root = '/data3/face_data/face_dataset_coco_positive/json/'
train_img_prefix = data_root + 'coco_face_train/'
val_img_prefix = data_root + 'coco_face_val/'
test_img_prefix = data_root + 'coco_face_val/'
train_ann_file = data_root + 'coco_face_train.json'
val_ann_file = data_root + 'coco_face_val.json'
test_ann_file = data_root + 'coco_face_val.json'

# Model
conv_cfg = dict(type='ConvDWS')
norm_cfg = dict(type='SyncBN', eps=1e-3)
act_cfg = dict(type='Swish')

load_from = None
resume_from = None
work_dir = 'work_dirs/atss_effdet_d0/'

model = dict(
	type='RetinaNet',
	pretrained=None,
	backbone=dict(
		type='TimmBackbone',
		model_name='efficientnet_b0',
		norm_eval=True,
		frozen_stages=-1,
		pretrained=True,
		drop_rate=0.0,
		drop_path_rate=0.1,
		pad_type='same',
	),
	neck=dict(
		type='BiFPN',
		in_channels=[24,40,112,320],
		out_channels=64,
		start_level=1,
		num_outs=5,
		stack=3,
		add_extra_convs=True,
		extra_convs_on_inputs=True,
		conv_cfg=conv_cfg,
		norm_cfg=norm_cfg,
		activation=act_cfg,
	),
	bbox_head=dict(
		type='ATSSEffDetHead',
		num_classes=2,
		in_channels=64,
		stacked_convs=3,
		num_levels=5,
		feat_channels=64,
		octave_base_scale=8,
		scales_per_octave=1,
		anchor_ratios=[1.0],
		anchor_strides=[8, 16, 32, 64, 128],
		target_means=[.0, .0, .0, .0],
		target_stds=[0.1, 0.1, 0.2, 0.2],
		conv_cfg=conv_cfg,
		norm_cfg=norm_cfg,
		act_cfg=act_cfg,
		loss_cls=dict(type='FocalLoss', use_sigmoid=True, gamma=2.0, alpha=0.25, loss_weight=1.0),
		loss_bbox=dict(type='GIoULoss', loss_weight=2.0),
		loss_centerness=dict(type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
	),
)
#training and testing settings
train_cfg = dict(
	assigner=dict(type='ATSSAssigner', topk=9),
	allowed_border=-1,
	pos_weight=-1,
	debug=False)
test_cfg = dict(
	nms_pre=1000,
	min_bbox_size=0,
	score_thr=0.05,
	nms=dict(type='nms', iou_thr=0.5),
	max_per_img=100)
# dataset settings
dataset_type = 'CocoDataset'
img_norm_cfg = dict(
	mean=[123.675, 116.28, 103.53],
	std=[58.395, 57.12, 57.375],
	to_rgb=True)
train_pipeline = [
	dict(type='LoadImageFromFile'),
	dict(type='LoadAnnotations', with_bbox=True, with_mask=False),
	dict(type='Resize', img_scale=img_scale, keep_ratio=keep_ratio),
	dict(type='RandomFlip', flip_ratio=0.5),
	dict(type='Pad', size_divisor=size_divisor),
	dict(type='Normalize', **img_norm_cfg),
	dict(type='DefaultFormatBundle'),
	dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]
test_pipeline = [
	dict(type='LoadImageFromFile'),
	dict(
		type='MultiScaleFlipAug',
		img_scale=img_scale,
		flip=False,
		transforms=[
			dict(type='Resize', keep_ratio=keep_ratio),
			dict(type='RandomFlip'),
			dict(type='Normalize', **img_norm_cfg),
			dict(type='Pad', size_divisor=size_divisor),
			dict(type='ImageToTensor', keys=['img']),
			dict(type='Collect', keys=['img']),
		])
]
data = dict(
	imgs_per_gpu=imgs_per_gpu,
	workers_per_gpu=workers_per_gpu,
	train=dict(
		type='RepeatDataset',
		times=repeat_times,
		dataset=dict(
			type=dataset_type,
			ann_file=train_ann_file,
			img_prefix=train_img_prefix,
			pipeline=train_pipeline,
		),
	),
	val=dict(
		type=dataset_type,
		ann_file=val_ann_file,
		img_prefix=val_img_prefix,
		pipeline=test_pipeline,
	),
	test=dict(
		type=dataset_type,
		ann_file=test_ann_file,
		img_prefix=test_img_prefix,
		pipeline=test_pipeline,
	),
)
# optimizer
optimizer = dict(
	type='SGD',
	lr=lr_start, momentum=0.9, weight_decay=weight_decay,
)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))

# learning policy
lr_config = dict(
	policy='cosine', target_lr=lr_end, by_epoch=False,
	warmup='linear', warmup_iters=warmup_iters, warmup_ratio=warmup_ratio,
)
checkpoint_config = dict(interval=ckpt_interval)

# yapf:disable
log_config = dict(
	interval=log_interval,
	hooks=[
		dict(type='TextLoggerHook'),
	])
# yapf:enable
# runtime settings
device_ids = range(8)
dist_params = dict(backend='nccl')
log_level = 'INFO'
workflow = [('train', 1)]

**Comment by betterhalfwzm** [comment]
@thuyngch Can you give me some advice？

**Comment by thuyngch** [comment]
Hi @betterhalfwzm ,

You can change a small number of epochs, for example, 80 epochs.
The original takes 300 epochs to finish the cosine learning rate schedule.
You are at epoch 77, so the learning rate is still large to converge.
Load my released weights as the pretraining and finetune on a smaller number of epochs.

**Comment by betterhalfwzm** [comment]
> Hi @betterhalfwzm ,
> 
> You can change a small number of epochs, for example, 80 epochs.
> The original takes 300 epochs to finish the cosine learning rate schedule.
> You are at epoch 77, so the learning rate is still large to converge.
> Load my released weights as the pretraining and finetune on a smaller number of epochs.

All test images have same issue like follow image: one confidence is high,and others is close to 0.
I'm very confused. @thuyngch 
![image](https://user-images.githubusercontent.com/20449990/79847617-8f6f9400-83f2-11ea-9016-e36762cc2a35.png)

**Comment by betterhalfwzm** [comment]
If I block one of them, the other can detect it. I'm confused. @thuyngch 
![image](https://user-images.githubusercontent.com/20449990/79849548-30f7e500-83f5-11ea-8d45-2ffbb9ef18d2.png)
